src/sbd-pacemaker.c

```diff
@@ -94,18 +94,79 @@ mon_timer_reconnect(gpointer data)
 		cl_log(LOG_INFO, "CIB reconnect successful");
 	}
 
+	notify_parent();
 	return FALSE;
 }
 
 static void
 mon_cib_connection_destroy(gpointer user_data)
 {
 	if (cib) {
-		cib->cmds->signoff(cib);
-		set_servant_health(pcmk_health_transient, LOG_WARNING, "Disconnected from CIB");
-		timer_id_reconnect = g_timeout_add(reconnect_msec, mon_timer_reconnect, NULL);
+        int new_servant_health = pcmk_health_transient;
+        xmlNode *cib_copy = NULL;
+        pe_working_set_t data_set;
+        node_t *node;
+
+        /* use leftover info to check if we are running
+         * on a remote-node without active resources
+         */
+        if (current_cib) {
+            cib_copy = copy_xml(current_cib);
+            if (cli_config_update(&cib_copy, NULL, FALSE)) {
+                last_refresh = time(NULL);
+                set_working_set_defaults(&data_set);
+                data_set.input = cib_copy;
+                data_set.flags |= pe_flag_have_stonith_resource;
+                cluster_status(&data_set);
+                compute_status(&data_set);
+                node = pe_find_node(data_set.nodes, local_uname);
+                if ((node != NULL) &&
+                    (node->details->type == node_remote) &&
+                    (node->details->running_rsc == NULL)) {
+
+                    if (data_set.no_quorum_policy != no_quorum_suicide) {
+                        new_servant_health = pcmk_health_online;
```

Comment:
I'd be wary of re-using "online" here, pcmk_health_pending might be better or invent a new one that means "safely disconnected dont shoot!"